### PR TITLE
Updated `setPusher` function 

### PR DESCRIPTION
**Pull request by Velin92**
Updated the set pusher function to work again (will only work once my [Rust PR](https://github.com/matrix-org/matrix-rust-sdk/pull/1608) is merged and released)

### CODE REVIEW

File: ElementX/Sources/Application/AppCoordinator.swift [hunk 1]
```diff
@@ -473,11 +473,11 @@ extension AppCoordinator: NotificationManagerDelegate {
     func notificationTapped(_ service: NotificationManagerProtocol, content: UNNotificationContent) async {
         MXLog.info("[AppCoordinator] tappedNotification")
 
-        guard let roomId = content.userInfo[NotificationConstants.UserInfoKey.roomIdentifier] as? String else {
+        guard !content.threadIdentifier.isEmpty else {
```
Comment (pixlwave) [inline ElementX/Sources/Application/AppCoordinator.swift:476]: Nice! Could we add a small doc comment to clarify that this is the notification's `thread-id` and nothing todo with matrix threads? I was a bit confused by this change at first 🙈

File: ElementX/Sources/Mocks/Generated/GeneratedMocks.swift [hunk 2]
```diff
@@ -11,18 +10,20 @@ class BugReportServiceMock: BugReportServiceProtocol {
...
         return crashCallsCount > 0
     }
     var crashClosure: (() -> Void)?
+
     func crash() {
         crashCallsCount += 1
         crashClosure?()
     }
-    // MARK: - submitBugReport
+    //MARK: - submitBugReport
```
Comment (pixlwave) [inline ElementX/Sources/Mocks/Generated/GeneratedMocks.swift:26]: Why the loss of the space here?

File: ElementX/Sources/Services/Client/ClientProxyProtocol.swift [hunk 3]
```diff
@@ -31,37 +31,21 @@ enum ClientProxyError: Error {
...
-//    }
-}
-
 enum SlidingSyncConstants {
     static let initialTimelineLimit: UInt = 0
     static let lastMessageTimelineLimit: UInt = 1
     static let timelinePrecachingTimelineLimit: UInt = 20
 }
 
+struct Pusher {
```
Comment (pixlwave) [inline ElementX/Sources/Services/Client/ClientProxyProtocol.swift:40]: Could we add some docs to the one please? I don't think the name `Pusher` would be immediately obvious to someone who is new to the project, so would be nice to add a description.

**Review by pixlwave — COMMENTED** [review]
Looks great, nice simplification with the Rust types 👍

File: ElementX/Sources/Services/Client/ClientProxyProtocol.swift (hunk 3 above)
Comment (Velin92) [reply ElementX/Sources/Services/Client/ClientProxyProtocol.swift:40]: renaming it also to PusherConfiguration

Comment (pixlwave) [reply ElementX/Sources/Services/Client/ClientProxyProtocol.swift]: Nice, much cleaner 👏

**Review by pixlwave — APPROVED** [review]
🚢 it!

**Pull request merged**